### DISCUSSION TITLE
Is there a way to use cmd shell without the /D flag ?

### DISCUSSION THREAD

**Discussion by goanpeca** [community/community#25105] [Actions]
<p>By design shells are not reading on ~/.profile, ~/.bash_profile (<a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell</a>) but I can get around that by writing an action that writes (or appends) to these files and then do something like</p>
<pre><code>- name: Bash
  shell: bash -l {0}
  run: ...
</code></pre>
<p>The cmd shell runs <strong>%ComSpec% /D /E:ON /V:OFF /S /C “CALL “{0}””</strong> so that is doing</p>
<pre><code>/D Ignore registry AutoRun commands
</code></pre>
<p>Is there a way to override this behavior? I have tried variations of </p>
<pre><code>- name: Cmd.exe
  shell: '&lt;font color="#808080"&gt;&lt;strong&gt;&lt;font face="courier new,courier"&gt;%ComSpec% /E:ON /V:OFF /S /C "CALL "{0}""&lt;/font&gt;&lt;/strong&gt;&lt;/font&gt; run: ...
- name: Cmd.exe
shell: cmd&lt;font color="#808080"&gt;&lt;strong&gt;&lt;font face="courier new,courier"&gt; /E:ON /V:OFF /S /C "CALL "{0}""&lt;/font&gt;&lt;/strong&gt;&lt;/font&gt;
run: ...
- name: Cmd.exe
shell: cmd.exe&lt;font color="#808080"&gt;&lt;strong&gt;&lt;font face="courier new,courier"&gt; /E:ON /V:OFF /S /C "CALL "{0}""&lt;/font&gt;&lt;/strong&gt;&lt;/font&gt;
run: ...
</code></pre>
<p>Any help is appreciated, thanks!</p>

**Comment by BrightRan** [comment]
<p>There is not any easy way to globally ignore registry AutoRun commands.<br>
If you are using a self-hosted Windows runner, you may be able to manually disable the AutoRun commands from registry, but this method may bring some risks, incorrectly editing the registry may severely damage the system.</p>

**Comment by goanpeca** [comment]
<p>&gt;There is not any easy way to globally ignore registry AutoRun commands.</p>
<p>This is the opposite of what I am asking. cmd shells always ignore AutoRun commands and I want to know if it is possible to NOT ignore them.</p>
<p>Cheers.</p>

**Comment by dakale** [comment, marked as answer]
<p>I believe what you have posted is correct as far as the shell option goes. </p>
<p>I wrote a quick sample:</p>
<pre><code>on:
  push:
    paths:
    - .github/workflows/shell-cmd.yaml

jobs:
  bash:
    runs-on: windows-latest
    steps:
    - name: No quotes
      continue-on-error: true
      run: echo "Hello world"
      shell: cmd /C CALL {0}

    - name: Full quotes
      continue-on-error: true
      run: echo "Hello world"
      shell: cmd /E:ON /V:OFF /S /C "CALL "{0}""

    - name: Some quotes
      continue-on-error: true
      run: echo "Hello world"
      shell: cmd /C CALL "{0}"
</code></pre>
<p>To verify different syntaxes and they all work fine. What part of this is not working? The contents of the actual script Im sure are more complicated than Hello World, so the issue might be there and not the shell option itself </p>

**Comment by goanpeca** [comment]
<p>Hello!</p>
<p>Not sure why the different options did not work. Ended up using this one.</p>
<pre><code>cmd /C CALL "{0}"
</code></pre>
<p>Cheers</p>
